[Attack discovery] Optionally update the `kibana.alert.workflow_status` of alerts associated with Attack discoveries

andrew-goldstein opened the pull request:
## [Attack discovery] Optionally update the `kibana.alert.workflow_status` of alerts associated with Attack discoveries

This PR introduces a new UI to optionally update the `kibana.alert.workflow_status` of alerts associated with Attack discoveries, as illustrated by the animated gif below:

![update_attack_discovery_alerts](https://github.com/user-attachments/assets/5974fc4e-50b4-43e9-9939-885f9577c153)

Users may (optionally) update all alerts for a single attack discovery, or just update the discovery itself:

![update_one_attack_discovery](https://github.com/user-attachments/assets/fd774ae7-976d-4649-a97d-b9bae8d359ad)

When multiple attack discoveries are selected, users may also (optionally) update the status of all their related alerts via the bulk action menu:

![update_multiple_discoveries](https://github.com/user-attachments/assets/71463945-f201-4810-9798-2646751dc919)

### Alert document enhancements

Attack discoveries generated via the Attack discovery page, and scheduled Attack discoveries (generated via the alerting framework), are persisted as alert documents.

To support the new UI, this PR populates Attack discovery alert documents with two additional (existing, but unused by Attack discovery) alert document fields:

1) `kibana.alert.start` - timestamp when Attack discoveries are created

2) `kibana.alert.workflow_status_updated_at` - timestamp when the `kibana.alert.workflow_status` was last updated

This PR introduces three new alert document fields to capture metadata about when alerts are updated. Attack discovery is the first implementation to use these new fields, however any consumer of the alerting framework may utilize them in the future:

1) `kibana.alert.updated_at` - timestamp when the alert was last updated

2) `kibana.alert.updated_by.user.id` - user id of the user that last updated the alert

3) `kibana.alert.updated_by.user.name` -  user name of the user that last updated the alert

The three new alert fields above are updated when Attack discovery users update:

- The `kibana.alert.workflow_status` status of Attack discoveries
- The visibility (sharing) status of Attack discoveries (`kibana.alert.attack_discovery.users`)

The three new fields above were added to the [alert_field_map](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-alerts-as-data-utils/src/field_maps/alert_field_map.ts) and [alert_schema](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-alerts-as-data-utils/src/schemas/generated/alert_schema.ts).

Using the `default` space as an example, the fields above may be observed in the `.adhoc.alerts-security.attack.discovery.alerts-default` data stream for Attack discoveries generated on the Attack discovery page, and scheduled discoveries for the same space are found in the `.alerts-security.attack.discovery.alerts-default` data stream.

### @timestamp updated when sharing status changes

To ensure newly-shared Attack discoveries are bumped to the top of search results, the `@timestamp` field is updated when the visibility (sharing) status of Attack discoveries (`kibana.alert.attack_discovery.users`) is updated.

(The original time an Attack discovery was generated is represented by the `kibana.alert.start` field, which is not mutated.)

### Visibility menu changes

This PR disables the visibility menu items for shared Attack discoveries, as illustrated by the screenshot below:

![visibility_menu_disabled](https://github.com/user-attachments/assets/168db75c-de8f-4bf1-9490-7e3995faed9d)

The disabled menu has a tooltip that reads:

```
The visibility of shared discoveries cannot be changed
```

Note: The internal Attack discovery bulk API still (intentionally) allows changes to the visibility of shared attack discoveries.

### `kibana.alert.workflow_status` added to default `Alerts` tab columns

The `kibana.alert.workflow_status` field was added to default `Alerts` tab columns, as illustrated by the screenshot below:

![alerts_tab_workflow_status](https://github.com/user-attachments/assets/264647d0-5782-444f-ad0e-c5485fae1e96)

### Summary of field updates

The following table describes when fields are updated (via this PR):

| Field                                     | Updated when                                                                                                                         | Description                                                                                  |
|-------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------|
| `kibana.alert.start`                      | - Attack discoveries are created                                                                                                     | timestamp when Attack discoveries are created                                                |
| `kibana.alert.workflow_status_updated_at` | - Workflow status (`kibana.alert.workflow_status`) is updated                                                                        | timestamp when `kibana.alert.workflow_status` was last updated                               |
| `kibana.alert.updated_at`                 | - Workflow status (`kibana.alert.workflow_status`) is updated<br>- Sharing status (`kibana.alert.attack_discovery.users`) is updated | timestamp when the alert was last updated                                                    |
| `kibana.alert.updated_by.user.id`         | - Workflow status (`kibana.alert.workflow_status`) is updated<br>- Sharing status (`kibana.alert.attack_discovery.users`) is updated | user id of the user that last updated the alert                                              |
| `kibana.alert.updated_by.user.name`       | - Workflow status (`kibana.alert.workflow_status`) is updated<br>- Sharing status (`kibana.alert.attack_discovery.users`) is updated | user name of the user that last updated the alert                                            |
| `@timestamp`                              | - Attack discoveries are created<br>- Sharing status (`kibana.alert.attack_discovery.users`) is updated                              | ECS [`@timestamp`](https://www.elastic.co/docs/reference/ecs/ecs-base#field-timestamp) field |

### Feature flags

The _required_ feature flag below is necessary to desk test with Ad hoc attack discoveries. The _recommended_ feature flag below enables testing with scheduled Attack discoveries.

### required: `securitySolution.attackDiscoveryAlertsEnabled`

Enable the required `securitySolution.attackDiscoveryAlertsEnabled` feature flag in `config/kibana.dev.yml`:

```yaml
feature_flags.overrides:
  securitySolution.attackDiscoveryAlertsEnabled: true
```

### recommended: `securitySolution.assistantAttackDiscoverySchedulingEnabled: true`

Also enable the recommended `assistantAttackDiscoverySchedulingEnabled` feature flag in `config/kibana.dev.yml`:

```yaml
feature_flags.overrides:
  securitySolution.attackDiscoveryAlertsEnabled: true
  securitySolution.assistantAttackDiscoverySchedulingEnabled: true
```


<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

andrew-goldstein committed:
## [Attack discovery] Optionally update the `kibana.alert.workflow_status` of alerts associated with Attack discoveries

This PR introduces a new UI to optionally update the `kibana.alert.workflow_status` of alerts associated with Attack discoveries, as illustrated by the animated gif below:

![update_attack_discovery_alerts](https://github.com/user-attachments/assets/5974fc4e-50b4-43e9-9939-885f9577c153)

Users may (optionally) update all alerts for a single attack discovery, or just update the discovery itself:

![update_one_attack_discovery](https://github.com/user-attachments/assets/fd774ae7-976d-4649-a97d-b9bae8d359ad)

When multiple attack discoveries are selected, users may also (optionally) update the status of all their related alerts via the bulk action menu:

![update_multiple_discoveries](https://github.com/user-attachments/assets/71463945-f201-4810-9798-2646751dc919)

### Alert document enhancements

Attack discoveries generated via the Attack discovery page, and scheduled Attack discoveries (generated via the alerting framework), are persisted as alert documents.

To support the new UI, this PR populates Attack discovery alert documents with two additional (existing, but unused by Attack discovery) alert document fields:

1) `kibana.alert.start` - timestamp when Attack discoveries are created

2) `kibana.alert.workflow_status_updated_at` - timestamp when the `kibana.alert.workflow_status` was last updated

This PR introduces three new alert document fields to capture metadata about when alerts are updated. Attack discovery is the first implementation to use these new fields, however any consumer of the alerting framework may utilize them in the future:

1) `kibana.alert.updated_at` - timestamp when the alert was last updated

2) `kibana.alert.updated_by.user.id` - user id of the user that last updated the alert

3) `kibana.alert.updated_by.user.name` -  user name of the user that last updated the alert

The three new alert fields above are updated when Attack discovery users update:

- The `kibana.alert.workflow_status` status of Attack discoveries
- The visibility (sharing) status of Attack discoveries (`kibana.alert.attack_discovery.users`)

The three new fields above were added to the [alert_field_map](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-alerts-as-data-utils/src/field_maps/alert_field_map.ts) and [alert_schema](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-alerts-as-data-utils/src/schemas/generated/alert_schema.ts).

Using the `default` space as an example, the fields above may be observed in the `.adhoc.alerts-security.attack.discovery.alerts-default` data stream for Attack discoveries generated on the Attack discovery page, and scheduled discoveries for the same space are found in the `.alerts-security.attack.discovery.alerts-default` data stream.

### @timestamp updated when sharing status changes

To ensure newly-shared Attack discoveries are bumped to the top of search results, the `@timestamp` field is updated when the visibility (sharing) status of Attack discoveries (`kibana.alert.attack_discovery.users`) is updated.

(The original time an Attack discovery was generated is represented by the `kibana.alert.start` field, which is not mutated.)

### Visibility menu changes

This PR disables the visibility menu items for shared Attack discoveries, as illustrated by the screenshot below:

![visibility_menu_disabled](https://github.com/user-attachments/assets/168db75c-de8f-4bf1-9490-7e3995faed9d)

The disabled menu has a tooltip that reads:

```
The visibility of shared discoveries cannot be changed
```

Note: The internal Attack discovery bulk API still (intentionally) allows changes to the visibility of shared attack discoveries.

### `kibana.alert.workflow_status` added to default `Alerts` tab columns

The `kibana.alert.workflow_status` field was added to default `Alerts` tab columns, as illustrated by the screenshot below:

![alerts_tab_workflow_status](https://github.com/user-attachments/assets/264647d0-5782-444f-ad0e-c5485fae1e96)

### Feature flags

The _required_ feature flag below is necessary to desk test with Ad hoc attack discoveries. The _recommended_ feature flag below enables testing with scheduled Attack discoveries.

### required: `securitySolution.attackDiscoveryAlertsEnabled`

Enable the required `securitySolution.attackDiscoveryAlertsEnabled` feature flag in `config/kibana.dev.yml`:

```yaml
feature_flags.overrides:
  securitySolution.attackDiscoveryAlertsEnabled: true
```

### recommended: `securitySolution.assistantAttackDiscoverySchedulingEnabled: true`

Also enable the recommended `assistantAttackDiscoverySchedulingEnabled` feature flag in `config/kibana.dev.yml`:

```yaml
feature_flags.overrides:
  securitySolution.attackDiscoveryAlertsEnabled: true
  securitySolution.assistantAttackDiscoverySchedulingEnabled: true
```

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

andrew-goldstein commented:
## Files by Code Owner

### #

* x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/table.tsx

### elastic/kibana-cases

* x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/table.tsx

### elastic/obs-ux-management-team

* src/platform/packages/shared/kbn-rule-data-utils/src/default_alerts_as_data.ts

### elastic/response-ops

* src/platform/packages/shared/kbn-alerts-as-data-utils/src/field_maps/alert_field_map.ts
* src/platform/packages/shared/kbn-alerts-as-data-utils/src/schemas/generated/alert_schema.ts
* src/platform/packages/shared/kbn-alerts-as-data-utils/src/schemas/generated/security_schema.ts
* src/platform/packages/shared/kbn-rule-data-utils/src/default_alerts_as_data.ts

### elastic/security-detections-response

* src/platform/packages/shared/kbn-rule-data-utils/src/default_alerts_as_data.ts

### elastic/security-generative-ai

* x-pack/platform/packages/shared/kbn-elastic-assistant-common/impl/schemas/attack_discovery/attack_discovery_alert.gen.ts
* x-pack/platform/packages/shared/kbn-elastic-assistant-common/impl/schemas/attack_discovery/attack_discovery_alert.schema.yaml
* x-pack/solutions/security/plugins/elastic_assistant/server/__mocks__/attack_discovery_alert_document_response.ts
* x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/get_update_attack_discovery_alerts_query/index.test.ts
* x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/get_update_attack_discovery_alerts_query/index.ts
* x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/persistence/transforms/transform_search_response_to_alerts/index.test.ts
* x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/persistence/transforms/transform_search_response_to_alerts/index.ts
* x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/persistence/transforms/transform_to_alert_documents/index.test.ts
* x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/persistence/transforms/transform_to_alert_documents/index.ts
* x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/schedules/fields/field_map.ts
* x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/schedules/types.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/panel_header/primary_interactions/badges/shared_badge/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/panel_header/primary_interactions/badges/shared_badge/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/panel_header/primary_interactions/badges/shared_badge/translations.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/ai_for_soc/table.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/ai_for_soc/wrapper.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/update_alerts_modal/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/update_alerts_modal/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/update_alerts_modal/translations.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/use_update_alerts_status/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/use_update_alerts_status/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/use_update_alerts_status/translations.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/use_attack_discovery_bulk/index.test.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/use_attack_discovery_bulk/index.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/use_attack_discovery_bulk/translations.ts

### elastic/security-solution

* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/panel_header/primary_interactions/badges/shared_badge/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/panel_header/primary_interactions/badges/shared_badge/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/panel_header/primary_interactions/badges/shared_badge/translations.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/ai_for_soc/table.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/ai_for_soc/wrapper.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/update_alerts_modal/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/update_alerts_modal/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/update_alerts_modal/translations.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/use_update_alerts_status/index.test.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/use_update_alerts_status/index.tsx
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/use_update_alerts_status/translations.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/use_attack_discovery_bulk/index.test.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/use_attack_discovery_bulk/index.ts
* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/use_attack_discovery_bulk/translations.ts
* x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/table.tsx
* x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/table/table.tsx

### elastic/security-threat-hunting-investigations

* x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/table.tsx
* x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/table/table.tsx

### remove

* x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/table.tsx

### soon

* x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/table.tsx

### to

* x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/table.tsx

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311953)
* Commit: cdacea75b446b6e5673e5d9922b6cf748338de1e

### Failed CI Steps
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f668-492e-b331-9921d34bd84d)
* [Jest Tests #19](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66c-45c3-998e-ca0a9c34d4ef)
* [Jest Tests #19](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a195-d000-4485-8434-18ef7973eb35)
* [Jest Integration Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169)
* [FTR Configs #100](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f6c9-4912-9846-2c269142615d)
* [FTR Configs #112](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f6d4-43cf-9f74-2a82f7148353)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66e-4e05-9a82-8502ab449169/artifacts/0197a19a-056f-49e8-bf06-c6748f4dd691) Jest Integration Tests #<span></span>7 / Alert as data fields checks detect AAD fields changes for: siem.eqlRule
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66e-4e05-9a82-8502ab449169/artifacts/0197a19a-056e-490c-8dc1-3eb0a1bd9b8f) Jest Integration Tests #<span></span>7 / Alert as data fields checks detect AAD fields changes for: siem.esqlRule
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66e-4e05-9a82-8502ab449169/artifacts/0197a19a-056e-49d0-9028-5b58ef31d1cd) Jest Integration Tests #<span></span>7 / Alert as data fields checks detect AAD fields changes for: siem.indicatorRule
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66e-4e05-9a82-8502ab449169/artifacts/0197a19a-056e-4153-9516-adf646a3bb99) Jest Integration Tests #<span></span>7 / Alert as data fields checks detect AAD fields changes for: siem.mlRule
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66e-4e05-9a82-8502ab449169/artifacts/0197a19a-056e-4c2f-8a5c-48ebd6682ca7) Jest Integration Tests #<span></span>7 / Alert as data fields checks detect AAD fields changes for: siem.newTermsRule
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66e-4e05-9a82-8502ab449169/artifacts/0197a19a-056e-433a-a377-501bb1b64c2b) Jest Integration Tests #<span></span>7 / Alert as data fields checks detect AAD fields changes for: siem.queryRule
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66e-4e05-9a82-8502ab449169/artifacts/0197a19a-056d-4be1-bc5e-67a50309d76f) Jest Integration Tests #<span></span>7 / Alert as data fields checks detect AAD fields changes for: siem.savedQueryRule
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66e-4e05-9a82-8502ab449169) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66e-4e05-9a82-8502ab449169/artifacts/0197a19a-056d-4456-a787-e753277ce1b4) Jest Integration Tests #<span></span>7 / Alert as data fields checks detect AAD fields changes for: siem.thresholdRule
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f6c9-4912-9846-2c269142615d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f6c9-4912-9846-2c269142615d/artifacts/0197a18f-740c-49df-85c7-564908d979bb) FTR Configs #<span></span>100 / Alerting alerts_as_data dynamic templates alerts as data fields limit should add the dynamic fields
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f6d4-43cf-9f74-2a82f7148353) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f6d4-43cf-9f74-2a82f7148353/artifacts/0197a18e-54e5-4e3e-af35-83db2d79909b) FTR Configs #<span></span>112 / Cloud Security Posture Test adding Cloud Security Posture Integrations CSPM AWS CIS_AWS Single Manual Temporary Keys CIS_AWS Single Manual Temporary Keys Workflow
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f668-492e-b331-9921d34bd84d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f668-492e-b331-9921d34bd84d/artifacts/0197a198-b9e7-4ee7-b147-785d8a4f42f7) Jest Tests #<span></span>9 / mappingFromFieldMap correctly creates mapping from field map
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a170-f66c-45c3-998e-ca0a9c34d4ef) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a170-f66c-45c3-998e-ca0a9c34d4ef/artifacts/0197a195-ba48-426a-8b39-7c79012b38c1) Jest Tests #<span></span>19 / matches snapshot
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311953#0197a195-d000-4485-8434-18ef7973eb35) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311953/jobs/0197a195-d000-4485-8434-18ef7973eb35/artifacts/0197a19c-542a-4d6f-ac50-fda402fc403a) Jest Tests #<span></span>19 / matches snapshot

cc @andrew-goldstein

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311953","number":311953,"commit":"cdacea75b446b6e5673e5d9922b6cf748338de1e"}],"number":311953}
buildkite-pr-comment-->

ymao1 reviewed (approved):
Response Ops changes LGTM. 3 new alert as data fields added. Code review only.

PhilippeOberti reviewed (changes requested):
This is a cool feature! Desk tested and things work nicely!

I left a couple of comments though: unless I'm mistaken, either some code is missing or we can undo the changes in a few files are they are not used anywhere.
If I'm correct and those changes are unused, you won't need approval from the @elastic/security-threat-hunting-investigations team anymore 😄

e40pud reviewed (approved):
Looks great! Thank you for pairing on review and testing it together 🚀

andrew-goldstein committed:
- removed outdated table changes (via PR feedback)

andrew-goldstein commented on x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/ai_for_soc/table.tsx:
++
Feedback addressed in commit https://github.com/elastic/kibana/pull/225029/commits/5681b3118b9275a32583588e3bea19aecb18a516

andrew-goldstein committed:
- restore `x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/table/table.tsx` (via PR feedback)

PhilippeOberti reviewed (approved):
Thanks for undoing the unnecessary changes!

dhurley14 reviewed (commented):
Fields need to be added to the alert schema for 8.19.

e40pud commented on src/platform/packages/shared/kbn-rule-data-utils/src/default_alerts_as_data.ts:
```diff
@@ -71,6 +71,15 @@ const ALERT_REASON = `${ALERT_NAMESPACE}.reason` as const;
 // kibana.alert.start - timestamp when the alert is first active
 const ALERT_START = `${ALERT_NAMESPACE}.start` as const;
 
+// kibana.alert.updated_at - timestamp when the alert was last updated
+const ALERT_UPDATED_AT = `${ALERT_NAMESPACE}.updated_at` as const;
+
+// kibana.alert.updated_by.user.id - user id of the user that last updated the alert
+const ALERT_UPDATED_BY_USER_ID = `${ALERT_NAMESPACE}.updated_by.user.id` as const;
+
+// kibana.alert.updated_by.user.name - user name of the user that last updated the alert
+const ALERT_UPDATED_BY_USER_NAME = `${ALERT_NAMESPACE}.updated_by.user.name` as const;
+
```
@andrew-goldstein it looks like we will need to add these common alert fields here https://github.com/elastic/kibana/blob/e3976c9c0f53cdeb6b4a8fdd05c9f691b898e120/x-pack/solutions/security/plugins/security_solution/common/api/detection_engine/model/alerts/8.19.0/index.ts#L32

andrew-goldstein committed:
- added new fields:

```
  ALERT_UPDATED_AT,
  ALERT_UPDATED_BY_USER_ID,
  ALERT_UPDATED_BY_USER_NAME,
```

to `x-pack/solutions/security/plugins/security_solution/common/api/detection_engine/model/alerts/8.19.0/index.ts`

dhurley14 reviewed (approved):
LGTM! Code review.

andrew-goldstein committed:
- fix for `Jest Tests #9 / mappingFromFieldMap correctly creates mapping from field map`

andrew-goldstein committed:
- fix for `Jest Tests #19 / matches snapshot`

andrew-goldstein committed:
- fix for `Jest Integration Tests #7 / Alert as data fields checks detect AAD fields changes for: siem.eqlRule`

andrew-goldstein committed:
- merge latest changes from `main` into `x-pack/platform/plugins/shared/alerting/server/integration_tests/__snapshots__/alert_as_data_fields.test.ts.snap`

andrew-goldstein committed:
- fixes tests in `x-pack/platform/test/alerting_api_integration/spaces_only/tests/alerting/group4/alerts_as_data/alerts_as_data_dynamic_templates.ts`

andrew-goldstein committed:
- adds new fields to `x-pack/platform/plugins/shared/alerting/server/integration_tests/__snapshots__/alert_as_data_fields.test.ts.snap`

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312306)
* Commit: de43a91bdacd1c5e53a0cbe6f3c562cf6d9cea82

### Failed CI Steps
* [FTR Configs #60](https://buildkite.com/elastic/kibana-pull-request/builds/312306#0197a39b-65b5-4519-bf09-97d1fc53251c)
* [FTR Configs #130](https://buildkite.com/elastic/kibana-pull-request/builds/312306#0197a39b-65f0-4353-87a7-935df07074c4)
* [FTR Configs #130](https://buildkite.com/elastic/kibana-pull-request/builds/312306#0197a3c5-fab5-487b-989d-37506e61ff25)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312306#0197a39b-65f0-4353-87a7-935df07074c4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312306/jobs/0197a39b-65f0-4353-87a7-935df07074c4/artifacts/0197a3c5-9650-445c-831c-7ad95638d220) FTR Configs #<span></span>130 / Alerting check alert schemas should not have discrepancies from the alert field map or the field map specific to a rule type
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312306#0197a3c5-fab5-487b-989d-37506e61ff25) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312306/jobs/0197a3c5-fab5-487b-989d-37506e61ff25/artifacts/0197a3dd-7ebf-4910-a111-c9ed8e3b35ed) FTR Configs #<span></span>130 / Alerting check alert schemas should not have discrepancies from the alert field map or the field map specific to a rule type
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312306#0197a39b-65b5-4519-bf09-97d1fc53251c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312306/jobs/0197a39b-65b5-4519-bf09-97d1fc53251c/artifacts/0197a3bd-c1c3-4ace-a58f-a19c153201e0) FTR Configs #<span></span>60 / InfraOps App Metrics UI Node Details #<span></span>Asset Type: host without metrics Overview Tab "before all" hook for "cpuUsage tile should be shown"

### History
* :broken_heart: [Build #311953](https://buildkite.com/elastic/kibana-pull-request/builds/311953) failed cdacea75b446b6e5673e5d9922b6cf748338de1e

cc @andrew-goldstein

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312306","number":312306,"commit":"de43a91bdacd1c5e53a0cbe6f3c562cf6d9cea82"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311953","number":311953,"commit":"cdacea75b446b6e5673e5d9922b6cf748338de1e"}],"number":312306}
buildkite-pr-comment-->

andrew-goldstein committed:
- updated `src/platform/packages/shared/kbn-alerts-as-data-utils/src/schemas/generated/streams_schema.ts` via output from running `node scripts/functional_test_runner --config=x-pack/platform/test/alerting_api_integration/spaces_only/tests/alerting/group4/config.ts --grep "check alert schemas"`

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312388)
* Commit: ebc1355cc62459dea5291a4e77c7cfc1ef165482

### Failed CI Steps
* [FTR Configs #125](https://buildkite.com/elastic/kibana-pull-request/builds/312388#0197a436-c82b-4512-95e0-354017f6b3ed)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312388#0197a436-c82b-4512-95e0-354017f6b3ed) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312388/jobs/0197a436-c82b-4512-95e0-354017f6b3ed/artifacts/0197a45b-f8c3-4592-870d-9a4486080de9) FTR Configs #<span></span>125 / Dataset Quality Dataset quality handles user privileges User can read logs-* User can monitor some data streams shows underprivileged warning when size cannot be accessed for some data streams

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/535c27fb9035895004585613c3b448714d42f594) | [after](https://github.com/elastic/kibana/commit/ebc1355cc62459dea5291a4e77c7cfc1ef165482) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7760 | 7764 | +4 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/535c27fb9035895004585613c3b448714d42f594) | [after](https://github.com/elastic/kibana/commit/ebc1355cc62459dea5291a4e77c7cfc1ef165482) | diff |
| --- | --- | --- | --- |
| `@kbn/rule-data-utils` | 185 | 191 | +6 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/535c27fb9035895004585613c3b448714d42f594) | [after](https://github.com/elastic/kibana/commit/ebc1355cc62459dea5291a4e77c7cfc1ef165482) | diff |
| --- | --- | --- | --- |
| `observability` | 1.3MB | 1.3MB | +132.0B |
| `securitySolution` | 9.7MB | 9.7MB | +6.1KB |
| `slo` | 991.8KB | 991.9KB | +132.0B |
| `synthetics` | 1.0MB | 1.0MB | +135.0B |
| total |  |  | +6.5KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/535c27fb9035895004585613c3b448714d42f594) | [after](https://github.com/elastic/kibana/commit/ebc1355cc62459dea5291a4e77c7cfc1ef165482) | diff |
| --- | --- | --- | --- |
| `observability` | 93.5KB | 93.7KB | +105.0B |
| `slo` | 32.5KB | 32.6KB | +105.0B |
| `synthetics` | 27.9KB | 28.0KB | +105.0B |
| `timelines` | 177.7KB | 177.9KB | +210.0B |
| total |  |  | +525.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/535c27fb9035895004585613c3b448714d42f594) | [after](https://github.com/elastic/kibana/commit/ebc1355cc62459dea5291a4e77c7cfc1ef165482) | diff |
| --- | --- | --- | --- |
| `@kbn/rule-data-utils` | 198 | 204 | +6 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/535c27fb9035895004585613c3b448714d42f594) | [after](https://github.com/elastic/kibana/commit/ebc1355cc62459dea5291a4e77c7cfc1ef165482) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 629 | 630 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/535c27fb9035895004585613c3b448714d42f594) | [after](https://github.com/elastic/kibana/commit/ebc1355cc62459dea5291a4e77c7cfc1ef165482) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 727 | 728 | +1 |


</details>


### History
* :broken_heart: [Build #312306](https://buildkite.com/elastic/kibana-pull-request/builds/312306) failed de43a91bdacd1c5e53a0cbe6f3c562cf6d9cea82
* :broken_heart: [Build #311953](https://buildkite.com/elastic/kibana-pull-request/builds/311953) failed cdacea75b446b6e5673e5d9922b6cf748338de1e

cc @andrew-goldstein

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312388","number":312388,"commit":"ebc1355cc62459dea5291a4e77c7cfc1ef165482"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312306","number":312306,"commit":"de43a91bdacd1c5e53a0cbe6f3c562cf6d9cea82"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311953","number":311953,"commit":"cdacea75b446b6e5673e5d9922b6cf748338de1e"}],"number":312388}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/15864637000

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/225349">](https://github.com/elastic/kibana/pull/225349)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
